gh-149763: Improve availablity docs in `select.rst` (#149764)
python/cpython: Doc/library/select.rst

   .. versionchanged:: 3.4
      The new file descriptor is now non-inheritable.

>    .. availability:: Solaris and derivatives.

Reviewer: Maybe this needs to be like this? If we need to mention derivatives, put it in the description?

```suggestion
   .. availability:: Solaris.
```

      When CPython is built, this function may be disabled using
      :option:`--disable-epoll`.

>    .. availability:: Linux 2.5.44 and newer.

Reviewer: ```suggestion
   .. availability:: Linux >= 2.5.44.
```

.. function:: poll()

   Returns a polling object, which
   supports registering and unregistering file descriptors, and then polling them
   for I/O events; see section :ref:`poll-objects` below for the methods supported
   by polling objects.

   Not supported by all operating systems.

> 

Reviewer 2: ```suggestion
```
I suggest removing this line. It is redundant now that the `.. availability:: Unix` directive has been added. Additionally, the phrasing 'Not supported by all...' is a bit vauge IMO.

   .. availability:: Unix.

.. function:: kqueue()

   Returns a kernel queue object; see section
   :ref:`kqueue-objects` below for the methods supported by kqueue objects.

   The new file descriptor is :ref:`non-inheritable <fd_inheritance>`.

   .. versionchanged:: 3.4
      The new file descriptor is now non-inheritable.

>    .. availability:: BSD and macOS.

Reviewer: ```suggestion
   .. availability:: BSD, macOS.
```

.. function:: kevent(ident, filter=KQ_FILTER_READ, flags=KQ_EV_ADD, fflags=0, data=0, udata=0)

   Returns a kernel event object; see section
   :ref:`kevent-objects` below for the methods supported by kevent objects.

>    .. availability:: BSD and macOS.

Reviewer: ```suggestion
   .. availability:: BSD, macOS.
```

.. _devpoll-objects:

> ``/dev/poll`` polling objects

Reviewer 2: ```suggestion
``/dev/poll`` Polling objects
```

.. _epoll-objects:

Edge and level trigger polling (epoll) objects
----------------------------------------------

   https://linux.die.net/man/4/epoll

>    *eventmask* parameter can be one of the following:

Reviewer 3: ```suggestion
   The *eventmask* is a bit mask using the following constants:
```

They can be combined, no?
